data/kernels/blendop.cl

```diff
@@ -580,49 +583,47 @@ blendop_Lab (__read_only image2d_t in_a, __read_only image2d_t in_b, __read_only
       o =  clamp((a * opacity) + (b * (1.0f - opacity)), min, max);
       break;
 
-    case DEVELOP_BLEND_BOUNDED:
-    case DEVELOP_BLEND_NORMAL:
-      o =  clamp((a * (1.0f - opacity)) + (b * opacity), min, max);
-      break;
-
     case DEVELOP_BLEND_LAB_LIGHTNESS:
     case DEVELOP_BLEND_LAB_L:
-      o.x = (a.x * (1.0f - opacity)) + (b.x * opacity);
-      o.y = a.y;
-      o.z = a.z;
+      o.x = clamp((a.x * (1.0f - opacity)) + (b.x * opacity), min.x, max.x);
+      o.y = clamp(a.y, min.y, max.y);
+      o.z = clamp(a.z, min.z, max.z);
       break;
 
     case DEVELOP_BLEND_LAB_A:
-      o.x = a.x;
-      o.y = (a.y * (1.0f - opacity)) + (b.y * opacity);
-      o.z = a.z;
+      o.x = clamp(a.x, min.x, max.x);
+      o.y = clamp((a.y * (1.0f - opacity)) + (b.y * opacity), min.y, max.y);
+      o.z = clamp(a.z, min.z, max.z);
       break;
 
     case DEVELOP_BLEND_LAB_B:
-      o.x = a.x;
-      o.y = a.y;
-      o.z = (a.z * (1.0f - opacity)) + (b.z * opacity);
+      o.x = clamp(a.x, min.x, max.x);
+      o.y = clamp(a.y, min.y, max.y);
+      o.z = clamp((a.z * (1.0f - opacity)) + (b.z * opacity), min.z, max.z);
       break;
 
     case DEVELOP_BLEND_LAB_COLOR:
-      o.x = a.x;
-      o.y = (a.y * (1.0f - opacity)) + (b.y * opacity);
-      o.z = (a.z * (1.0f - opacity)) + (b.z * opacity);
+      o.x = clamp(a.x, min.x, max.x);
+      o.y = clamp((a.y * (1.0f - opacity)) + (b.y * opacity), min.y, max.y);
+      o.z = clamp((a.z * (1.0f - opacity)) + (b.z * opacity), min.z, max.z);
       break;
       
     case DEVELOP_BLEND_HSV_LIGHTNESS:
     case DEVELOP_BLEND_HSV_COLOR:
     case DEVELOP_BLEND_RGB_R:
     case DEVELOP_BLEND_RGB_G:
     case DEVELOP_BLEND_RGB_B:
-      o = a;                            // Noop for Lab (without clamping)
+      o = clamp(a, min, max);
       break;
 
     /* fallback to normal blend */
+
+    case DEVELOP_BLEND_BOUNDED:
+    case DEVELOP_BLEND_NORMAL:
     case DEVELOP_BLEND_UNBOUNDED:
     case DEVELOP_BLEND_NORMAL2:
     default:
-      o =  (a * (1.0f - opacity)) + (b * opacity);
+      o =  clamp((a * (1.0f - opacity)) + (b * opacity), min, max);
```

Comment:
normal blend shouldn't clip

Reply:
Code is kept as is for Lab/raw/display RGB blending modes. Scene referred blending mode does not clip.